### PR TITLE
Added a function to truncate strings given a maximum number of characters with its test case

### PR DESCRIPTION
**Pull request by baimamboukar**
# Welcome to Dart community

### **Describe your change:**

* [✅ ] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [✅ ] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Dart/blob/master/CONTRIBUTING.md).
* [✅ ] This pull request is all my own work -- I have not plagiarized.
* [ ✅] I know that pull requests will not be merged if they fail the automated tests.
* [✅ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Dart files are placed inside an existing directory.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Review by akashgk — CHANGES_REQUESTED** [review]
Please add the tests to the same file instead of creating multiple files for the same.

**Comment by baimamboukar** [comment]
> Please add the tests to the same file instead of creating multiple files for the same. 

Done ✔️
I added the tests in the same file

File: strings/truncate_string.dart [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+void main() {
+  test(
+    "Truncate a long string given the max number of words and or suffix",
+    () {
+      String stringToTruncate =
+          "The beautiful thing with learning is that no one can take it away from you";
+      expect(truncateString(stringToTruncate, maxNumberOfWords: 40),
+          equals("The beautiful thing with learning is..."));
+    },
+  );
```
Comment (akashgk) [inline strings/truncate_string.dart:19]: Add more test cases. More Strings and edge cases.

**Review by akashgk — CHANGES_REQUESTED** [review]
LGTM, Please add more tests.

File: strings/truncate_string.dart (hunk 1 above)
Comment (baimamboukar) [reply strings/truncate_string.dart]: Hello @akashgk! 
Added more test cases...
Thanks for reviewing

File: strings/truncate_string.dart [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import 'package:test/test.dart';
+
+String truncateString(String string,
+    {int maxNumberOfWords, String suffix = '...'}) {
+  return (string.length - 3) < maxNumberOfWords
```
Comment (jerold) [inline strings/truncate_string.dart:5]: - Ternary checks the character length of the string, not the word count as input parameters suggest.
- subtracting 3 from `string.length` seems to try and account for the length of the suffix, but if a different suffix is supplied, it's length may not be 3.
- If the string's number of words is less than the maxNumberOfWords, you won't have performed a truncation, and will not need the suffix/replacement. ("..." at the end implies there is more to the string's content, but this would not be true if nothing is cut off the end)

**Comment by Coded-t** [comment]
please add more test
